src/components/navigation/link.ts

```diff
@@ -117,8 +121,17 @@ export class Link {
       return false;
     }
 
+    if (
+      userType === UserType.Guest &&
+      visibilitySettings.userType &&
+      visibilitySettings.userType[userType] !== undefined
+    ) {
+      return !!visibilitySettings.userType[userType];
+    }
```

Comment:
Why do we need an additional condition, shouldn't the guest case match in the other condition with defaulting an unauthenticated user to guest

Reply:
We have one header item, where we need to link to different URL based on user type (private/professional) on one brand (motoscout24). 
We will pass first check because brand check will return `false`
``` 
if (!visibilitySettings.brand[brand]) {
      return false;
}
```
We will pass second check because userType will be `undefined`, and at the end both links will be visible.
```
 if (
    userType &&
    visibilitySettings.userType &&
    !visibilitySettings.userType[userType]
  ) {
    return false;
  }
```    

Thats why I had to introduce new check for `guest` userType. With this check I can control which link won't be visible for unauthenticated users.

This was old implementation. If you asked about new implementation for link visibility, we need additional check because without it, both links for private and professional users will be visible.
```
userType &&
userType !== UserType.Guest &&
visibilitySettings.userType &&
!visibilitySettings.userType[userType]
```
if userType is `guest` or `undefined` here, `determineVisibility` will return `true` for both header items we defined in headerConfig.

Maybe you didn't want detailed explanation like this, but it helped me verify my solution :D